x-pack/legacy/plugins/infra/docs/test_setups/infra_metricbeat_aws.md

```diff
@@ -0,0 +1,85 @@
+- Navigate to the EC2 service, and create an instance. For everything in this guide, any of the free-tier instances with a plain linux installation will work fine.
+- Log into your new AWS instance, install metricbeat, and configure it to send data to your `elasticsearch` installation or cloud deployment. By default only the `system` module will be activated in `metricbeat`, this is fine. Verify that `add_cloud_metadata` is enabled in `metricbeat.yml`. (See also: https://www.elastic.co/guide/en/beats/metricbeat/current/add-cloud-metadata.html )
+- Start `metricbeat`
+- Login to Kibana and verify in Discover that the documents from your new AWS instance arrive, and include cloud metadata looking like this:
+```
+"cloud": {
+      "instance": {
+        "id": "i-011454f72559c510b"
+      },
+      "machine": {
+        "type": "t2.micro"
+      },
+      "region": "us-east-2",
+      "availability_zone": "us-east-2c",
+      "provider": "aws"
+    }
+```
+
+## AWS IAM User and IAM Policy
+
+In order to connect with Metricbeat's `aws` module to AWS and collect data, we need a user account with the correct access privileges. How to do this is described in https://www.elastic.co/blog/monitoring-aws-ec2-using-metricbeat-and-the-elastic-stack , or follow the steps below.
+
+First, we need an IAM Policy.
+
+- Navigate to the IAM (Identity and Access Management) service in the AWS management console.
+- Navigate to the "Policies" section.
+- Click "Create Policy".
+- Add four permissions to the new policy: `ec2:DescribeInstances`, `ec2:DescribeRegions`, `cloudwatch:GetMetricData`, `cloudwatch:ListMetrics`.
+- Give the new policy a name and save it.
+
+Second, we need an IAM User.
+
+- Navigate to the "Users" section in the IAM Service in the AWS management console.
+- Click "Add user".
+- Give it a name, and choose the access type "Programmatic Access".
+- On the "Set permissions" screen, choose "Attach existing policies directly", then find and check your newly created policy from the list.
+- Don't do anything in the "tags" step.
+- After the user has been created, copy or download the access key ID and the secret access key for the new user, before navigating away from the page (they will only be shown once).
+
+## Monitoring host
+
+How to set up the monitoring host (this can be your laptop or local workstation) is also described in https://www.elastic.co/blog/monitoring-aws-ec2-using-metricbeat-and-the-elastic-stack .
+
+In summary:
+
+- Install `metricbeat` and configure it to send data to your `elasticsearch` instance or cloud deployment.
+- Enable the `aws` module with `./metricbeat module enable aws`.
+- In `modules.d/aws.yml`, you can disable (comment out) all sections for metricsets other than `ec2`. Also, you don't need to configure the `session_token` if you use `access_key_id` and `secret_access_key`
+- (Note: my instance is in region `us-east-2c` according the the AWS console, but I had to configure `us-east-2` in `modules.d/aws.yml`.)
```

Comment:
Is it `default_region` or `regions` you configured here?

Reply:
This was for `default_region`.

Reply:
Gotcha, I believe you don't need to manually set `default_region`. That is used just to start aws api calls. If you want to limit aws module to only collect metrics from one or more specific regions, then you can use `regions` in the config". For example:
```
- module: aws
  period: 300s
  metricsets:
    - ec2
  access_key_id: '${AWS_ACCESS_KEY_ID:""}'
  secret_access_key: '${AWS_SECRET_ACCESS_KEY:""}'
  session_token: '${AWS_SESSION_TOKEN:""}'
  regions:
    - us-east-1
    - us-west-1
```
This will make aws module to collect only from `us-east-1` and `us-west-1` regions.